src/UhdmAst.cpp

```diff
@@ -487,6 +492,25 @@ AstNodeDType* getDType(FileLine* fl, vpiHandle obj_h, UhdmShared& shared) {
         if (dtype == nullptr) { dtype = basic; }
         break;
     }
+    case vpiArrayTypespec: {
+        AstRange* rangeNodep = nullptr;
+        std::stack<AstRange*> range_stack;
+        visit_one_to_many({vpiRange}, obj_h, shared, [&](AstNode* node) {
+            rangeNodep = reinterpret_cast<AstRange*>(node);
+            range_stack.push(rangeNodep);
+        });
+
+        auto elem_typespec_h = vpi_handle(vpiElemTypespec, obj_h);
+        AstNodeDType* dtypep = getDType(fl, elem_typespec_h, shared);
```

Comment:
We already have `AstNodeDType* dtype` here - which, as a matter of fact, should have been called `dtypep`. Can we just use that (and maybe rename it)?

Reply:
Done, I renamed it in the whole function